modules/ui/fields/input.js

```diff
@@ -167,9 +168,29 @@ export function uiFieldText(field, context) {
                     if (value) window.open(value, '_blank');
                 })
                 .merge(outlinkButton);
+        } else if (field.key.includes('colour')) {
+            input.attr('type', 'text');
+
+            updateColourPreview();
         }
     }
 
+    function updateColourPreview() {
+        wrap.selectAll('.foreign-id-permalink')
+            .remove();
+
+        outlinkButton = wrap.selectAll('.foreign-id-permalink')
+            .data([utilGetSetValue(input)], d => d);
```

Comment:
Does this do any validation? [This documentation](https://wiki.openstreetmap.org/wiki/Key:colour) allows `#XXXXXX` and CSS color keywords, but other [CSS color](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) formats like `rgb()`, `hsl()`, and `rgba()` are excluded from that definition. A working color well would mislead the user into thinking an invalid value is valid.

Reply:
good point, I've added validation based on what [openstreetmap-website does](https://github.com/openstreetmap/openstreetmap-website/blob/08e2a0/app/helpers/browse_tags_helper.rb#L177-L189).

They also check the colours against a list of all possible colour names, but we don't need to do that since setting an invalid background-colour value will be ignored by the browser